pkg/apis/servicecatalog/checksum/checksum_test.go

```diff
@@ -64,3 +64,19 @@ func TestBindingSpecChecksum(t *testing.T) {
 		t.Fatalf("versioned and unversioned checksums should match; expected %v, got %v", e, a)
 	}
 }
+
+func TestBrokerSpecChecksum(t *testing.T) {
+	spec := servicecatalog.BrokerSpec{
+		URL:      "https://kubernetes.default.svc:443/brokers/template.openshift.io",
+		AuthInfo: nil,
+	}
+
+	unversionedChecksum := unversioned.BrokerSpecChecksum(spec)
+	versionedSpec := v1alpha1.BrokerSpec{}
+	v1alpha1.Convert_servicecatalog_BrokerSpec_To_v1alpha1_BrokerSpec(&spec, &versionedSpec, nil /* conversionScope */)
+	versionedChecksum := checksumv1alpha1.BrokerSpecChecksum(versionedSpec)
+
+	if e, a := unversionedChecksum, versionedChecksum; e != a {
+		t.Fatalf("versioned and unversioned checksums should match; expected %v, got %v", e, a)
```

Comment:
Yes, they should match, and if we have multiple API versions at the same time, all API versions and unversioned must match.  They have to match because the API server calculates the checksum using the internal API and the controller does it using the versioned API.